lib/rules/jsx-key.js

```diff
@@ -61,9 +68,34 @@ module.exports = {
       return body.filter((item) => item.type === 'ReturnStatement')[0];
     }
 
+    function isKeyAfterSpread(attributes) {
+      let keyAttributeIndex = -1;
+      let spreadAttributeIndex = -1;
+
+      attributes.forEach((attribute, i) => {
+        if (attribute.type === 'JSXSpreadAttribute') {
+          spreadAttributeIndex = i;
+        }
+
+        if (attribute.type === 'JSXAttribute') {
+          if (propName(attribute) === 'key') {
+            keyAttributeIndex = i;
+          }
+        }
+      });
+
+      return spreadAttributeIndex !== -1 && keyAttributeIndex > spreadAttributeIndex;
+    }
+
     return {
       JSXElement(node) {
         if (hasProp(node.openingElement.attributes, 'key')) {
+          if (checkKeyMustBeforeSpread && isKeyAfterSpread(node.openingElement.attributes)) {
+            context.report({
+              node,
+              message: '"key" prop must before "...spread"'
```

Comment:
```suggestion
              message: '`key` prop must before any `{...spread}, to avoid conflicting with React’s new JSX transform: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html`'
```

Reply:
updated.